sdk/src/types/block/slot/epoch.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use derive_more::{Deref, Display, From, FromStr};
+
+use super::SlotIndex;
+use crate::types::block::Error;
+
+/// The tangle timeline is divided into epochs, and each epoch has a corresponding epoch index. Epochs are further
+/// subdivided into slots, each with a [`SlotIndex`].
+/// To calculate the epoch index of a timestamp, `slotsPerEpochExponent` and `slotDurationInSeconds` are needed.
+/// An epoch consists of `2^slotsPerEpochExponent` slots.
```

Comment:
Maybe add the examples from https://github.com/iotaledger/iota.go/commit/15e523d89ae481033f18059caed1490c7b812da3 as well

Reply:
According to this example, epoch 0 isn't reserved?

```
// epoch 0: <slot 0; slot 8)
// epoch 1: <slot 8; slot 16)
// epoch 2: <slot 16; slot 24)
```

Reply:
Yes only slot index 0 is reserved for genesis

Reply:
I suggested it before this iota.go commit, so I guess it changed and is now not correct anymore

Reply:
Asked here https://github.com/iotaledger/tips-draft/pull/20/files#r1305648933 to have it clarified

Reply:
done